### DISCUSSION TITLE
good quality audio files customized in flac preprocessing conversions (Update)

### DISCUSSION THREAD

**Discussion by impotentfield** [enzo1982/freac#404] [Audio Format Discussion]
i might have made a big mistake trying to share a file over the discussion forum. basically i want to make known that certain modification of the original flac file can alter the state of quality of the result conversion when listening. certain details can be picked out of the percussion and acoustics when applying varying functions in just the right way. when i started to experiment with flac i apply a conversion, listened with my walkman, then kept applying deeper and more complex applications in the apodization window where i listened over the results and made further assumptions where the apod functions were to be applied in more correct order whereas the order being specific to what function does what before or after became to me as a quality alteration. if one person at least wanted to apply apod steps in converting a file i would recommend using a certain proposed order of functions for converting in such a certain way to have a accurate quality in the result file. if one person then replied to himself and to me that there was a significance then altogether others would assume the position of realizing that the flac format allows the modification for which the result being witnessed while listening contains the main point in how flac files are to be part of your listening experience. in this manner of which people listen and the main point is quality in the result where flac allows correcting the details of the file when there is listening then converting then listening and so forth, the idea is to issue the process of communication where there might be misinterpretation of how to maintain that flac files have their remaining complexity on the side of the format preprocessing steps. altogether if people want to hear good music, and have no clue about the complex steps to maintain quality in the result then there would be chaos in their experience, whereas i issue my experiments in communicating the results as i go through my working out the preprosssing myself all the same comes out to make the idea aware that flac allows correcting the original file for high quality experience.

**Comment by Lee-Carre** [comment]
Previously: #445.

Paragraphs help readability.

Again, FLAC is lossless (decoded waveform matches the original input waveform). So, unless this is purely a psych phenomenon, then you've got some kind of hardware problem.

Have you updated the firmware on your portable player?

> certain details can be picked out of the percussion and acoustics when applying varying functions in just the right way.

Have you performed (mathematical) acoustic analysis (such as frequency-spectrum analysis), to corroborate this?

When you run the encoder with verification enabled, does it throw errors?

> where the apod functions were to be applied in more correct order whereas the order being specific to what function does what before or after

This is incorrect, to my understanding of libFLAC's implementation. The semi-colon separated list of functions are alternates for the encoder to try, and use the best result from. They're not a sequential chain. The idea is more granular control; trying more functions (or the same with different parameters) adds compute time for potentially greater compression, whereas fewer is less number-crunching for the encoder to do at the potential expense of sub-optimal compression.

The order of functions (assuming it's used literally, rather than parsed into an array and executed in some other order) simply means which of the alternatives the encoder tries first/later (so, you might want the computationally-cheap (or least memory-intense) ones first); but they're each treated as a candidate for comparison, rather than a chain.
If you're going for maximal compression (regardless of CPU time), then you'd stick a whole bunch (up to 32) in there, for the encoder to try.

Unless you're using it in some unusual way?

Else, if I'm completely wrong, then please correct me with a pointer to explanatory documentation. Though, the man page makes quite clear that multiple `-A` arguments may be passed, in any order.

I'm also now thinking of questions related to the source you're having the FLAC encoder use as input. Care to describe it?
